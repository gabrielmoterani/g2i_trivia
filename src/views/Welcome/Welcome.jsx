// React Imports
import React from 'react'
import { StyleSheet, Text, View, TouchableOpacity } from 'react-native';


// Libs Import
import { LinearGradient } from 'expo-linear-gradient';
import SafeAreaView from 'react-native-safe-area-view';


//Static Imports 
import COLORS from '../../config/theme';

const Welcome = ({navigation}) => {
    return (       
    
    <LinearGradient style={styles.container} colors={COLORS.gradients[0]}>
      <SafeAreaView style={styles.container}>
        <View style={styles.welcomeContainer}>
          <Text style={styles.welcomeText}>Welcome to the Trivia Chalenge!</Text>
        </View>
        <View style={styles.description}>
          <Text style={styles.descriptionText}>
            You will be presented with 
             <Text style={styles.true}> True </Text>
             or 
             <Text style={styles.false}> False </Text> 
            questions.
            </Text>
        </View>
        <View style={styles.description}>
          <Text style={styles.descriptionText}>
            Can you score 100%?
          </Text>
        </View>
        <View style={styles.btnContainer}>
          <TouchableOpacity
          onPress={() => {navigation.push('Quiz')}}
          style={styles.btn}>
            <Text style={styles.btnText}>
              Begin
            </Text>
          </TouchableOpacity>
        </View>
      </SafeAreaView>
    </LinearGradient > 
      
    );
}

const styles = StyleSheet.create({
    container: {
      flex: 1,
      alignItems: 'center',
      paddingVertical: 20
    },
    welcomeContainer:{
      flex: 1,
      marginTop: 30,
      marginLeft: 10,
      alignSelf: 'flex-start',
    },
    welcomeText: {
      fontSize: 30,
      fontWeight: 'bold',
      textAlign: 'left',
      width: 250,
      color: COLORS.primary
    },
    description:{
      flex: 0.5,
    },
    descriptionText:{
      fontSize: 20,
      fontWeight: 'bold',
      textAlign: 'center',
      marginHorizontal: 70,
      color: COLORS.primary,
    },
    true:{
      color: COLORS.green
    },
    false: {
      color: COLORS.pink
    },
    btnContainer: {},
    btn: {        
      backgroundColor: COLORS.background,
      paddingVertical: 10,
      borderRadius: 5,
    },
    btnText: {
      color: COLORS.primary,
      textAlign: 'center',
      textTransform: 'uppercase',
      fontWeight: 'bold',
      width: 220
    }
  });
 
export default Welcome;