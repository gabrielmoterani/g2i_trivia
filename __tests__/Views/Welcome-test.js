import React from 'react';
import { shallow } from 'enzyme';
import Welcome from '../../src/views/Welcome';

const setup = () => shallow(<Welcome />);


describe('Test Welcome Screen', () => {
    const wrapper = setup();
  
    it('should render properly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  
    it('should have a button', () => {
      const btns = wrapper.find('TouchableOpacity');
      expect(btns).toExist()
    }); 
  
  });