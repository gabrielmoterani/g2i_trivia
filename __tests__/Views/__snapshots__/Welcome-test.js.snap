// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Welcome Screen should render properly 1`] = `
<LinearGradient
  colors={
    Array [
      "#2C7EF8",
      "#2C7EF8",
      "#07ADD6",
    ]
  }
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "paddingVertical": 20,
    }
  }
>
  <SafeAreaView
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "paddingVertical": 20,
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "flex-start",
          "flex": 1,
          "marginLeft": 10,
          "marginTop": 30,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 30,
            "fontWeight": "bold",
            "textAlign": "left",
            "width": 250,
          }
        }
      >
        Welcome to the Trivia Chalenge!
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 0.5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 20,
            "fontWeight": "bold",
            "marginHorizontal": 70,
            "textAlign": "center",
          }
        }
      >
        You will be presented with
        <Text
          style={
            Object {
              "color": "#49dab4",
            }
          }
        >
           True 
        </Text>
        or
        <Text
          style={
            Object {
              "color": "#ff3859",
            }
          }
        >
           False 
        </Text>
        questions.
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 0.5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 20,
            "fontWeight": "bold",
            "marginHorizontal": 70,
            "textAlign": "center",
          }
        }
      >
        Can you score 100%?
      </Text>
    </View>
    <View
      style={Object {}}
    >
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Object {
            "backgroundColor": "#f00d00",
            "borderRadius": 5,
            "paddingVertical": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontWeight": "bold",
              "textAlign": "center",
              "textTransform": "uppercase",
              "width": 220,
            }
          }
        >
          Begin
        </Text>
      </TouchableOpacity>
    </View>
  </SafeAreaView>
</LinearGradient>
`;
